### PR TITLE
Bugfix: Runing periodic snapshot even if in analytical mode

### PR DESCRIPTION
**Pull request by andrejtonev**
### Description

Fixing unexpected behavior. 
Periodic snapshots were being created even in IN_MEMORY_ANALYTICAL mode.

[master < Task] PR
- [x] Provide the full content or a guide for the final git message
    - **Bugfix: Ran periodic snapshot in analytical mode**

### Documentation checklist
- [x] Add the documentation label tag
- [x] Add the bug / feature label tag
- [x] Add the milestone for which this feature is intended
    - If not known, set for a later milestone
- [x] Write a release note, including added/changed clauses
    - **Fixed periodic snapshots running in analytical mode**
- [x] Link the documentation PR here
    - **NA**
- [x] Tag someone from docs team in the comments

### CODE REVIEW

File: src/storage/v2/inmemory/storage.cpp [hunk 1]
```diff
@@ -2287,7 +2288,8 @@ void InMemoryStorage::CreateSnapshotHandler(
   };
 
   // Run the snapshot thread (if enabled)
-  if (config_.durability.snapshot_wal_mode != Config::Durability::SnapshotWalMode::DISABLED) {
+  if (storage_mode_ == StorageMode::IN_MEMORY_TRANSACTIONAL &&
+      config_.durability.snapshot_wal_mode != Config::Durability::SnapshotWalMode::DISABLED) {
```
[the remark this answers is not in the document]
Comment (andrejtonev) [reply src/storage/v2/inmemory/storage.cpp:2292]: This is the InMemory implementation.
For OnDisk we don't use our custom snapshots.
Not sure if we support periodic snapshots there. Probably not.

[the remark this answers is not in the document]
Comment (Ignition) [reply src/storage/v2/inmemory/storage.cpp:2292]: IMO do all the logic at config parsing time 
```cpp
  using namespace std::chrono_literals;
  using enum memgraph::storage::StorageMode;
  using enum memgraph::storage::Config::Durability::SnapshotWalMode;

  if (db_config.salient.storage_mode == IN_MEMORY_TRANSACTIONAL) {
    db_config.durability.snapshot_interval = std::chrono::seconds(FLAGS_storage_snapshot_interval_sec);

    if (db_config.durability.snapshot_interval == 0s) {
      db_config.durability.snapshot_wal_mode = DISABLED;
      LOG_FATAL(
          "In order to use write-ahead-logging you must enable "
          "periodic snapshots by setting the snapshot interval to a "
          "value larger than 0!");
    } else if (FLAGS_storage_wal_enabled) {
      db_config.durability.snapshot_wal_mode = PERIODIC_SNAPSHOT_WITH_WAL;
    } else {
      db_config.durability.snapshot_wal_mode = PERIODIC_SNAPSHOT;
    }
  } else {
    // IN_MEMORY_ANALYTICAL and ON_DISK_TRANSACTIONAL do not care
    db_config.durability.snapshot_wal_mode = DISABLED;
    db_config.durability.snapshot_interval = 0s;
  }
```

[the remark this answers is not in the document]
Comment (antoniofilipovic) [reply src/storage/v2/inmemory/storage.cpp:2292]: Makes sense

**Review by antoniofilipovic — APPROVED** [review]

**Comment by katarinasupe** [comment]
@andrejtonev does this close https://github.com/memgraph/memgraph/issues/1824?

**Comment by andrejtonev** [comment]
closes #1824

**Pull request merged**